library/core/class.dispatcher.php

```diff
@@ -720,7 +720,11 @@ private function rewriteRequest($request) {
         } elseif (in_array($request->path(), ['', '/'])) {
             $this->isHomepage = true;
             $defaultController = Gdn::router()->getRoute('DefaultController');
+            $originalGet = $request->get();
             $request->pathAndQuery($defaultController['Destination']);
+            if (is_array($originalGet) && !empty($originalGet)) {
+                $request->setQuery(array_merge($request->get(), $originalGet));
```

Comment:
We should be using `array_merge($originalGet, $request->get())` to preserve the DefaultController query string.

Reply:
That's going to lead to unexpected behavior. If I'm supplying something that's a "default", I wouldn't expect it to overwrite an explicit parameter being passed in. To a user, it would appear as if their query string was being ignored.

Reply:
The inverse is also true.
If you set a query string parameter in the default route it can be important. Letting someone overwrite it might not be a desired behaviour. It would be safer to not overwrite them. 

If there is a need to overwrite it I would argue that the caller should request the final path with the desired query string parameter.

Reply:
A non optimal setup could redirect the default page to a page that use a query parameter to do a subsequent redirect. Letting someone overwrite that parameter might lead to an unexpected behaviour.

Reply:
After discussing that the consensus is that we should not restrict anything because it would be weirdly used.